File: unit_03/w07_d2/lectures/ExpressTrain/app/controllers/trainsController.js
const Train = require('../models/Train');

const trainsController = {
  getAll(request, response) {
    Train.all().then((trains) => {
      response.status(200).send(trains); // SEND 200 oK... along with trains
    });
  },
  createTrain(request, response) {
    const trainData = request.body;
    Train.create(trainData).then((newTrain) => {
      response.status(200).send(newTrain);
    });
  },
  deleteTrain(request, response) {
    const id = request.params.id;
    Train.delete(id).then(() => {
      response.status(204).send();  // 204... no content
    });
  },
};

module.exports = trainsController;

File: unit_04/w10_d1/lab/NewYorkNewYork/app/controllers/trains_controller.rb
class TrainsController < ApplicationController
  # INDEX
  get "/" do
    trains = Train.all
    trains.to_json
  end

  # SHOW
  get "/:id" do
    train = Train.find(params[:id])
    train.to_json
  end
  
  # CREATE
  post "/" do
    new_train = Train.create(params[:train])
    new_train.to_json
  end

  # UPDATE
  patch "/:id" do
    train = Train.find(params[:id])
    train.update(params[:train])
    train.to_json
  end

  # DESTROY
  delete "/:id" do
    Train.delete(params[:id])
    status 204
  end
end
